src/helpers/http.js

```diff
@@ -47,12 +60,21 @@ function resolveHttpPromise(httpEvent, resolveFunction, startTime) {
  * @param {object} httpEvent The current event
  * @param {string} key name in metadata
  * @param {string} data data to jsonify
+ * @param {string} encoding data encoding from the headers
  */
-function setJsonPayload(httpEvent, key, data) {
+function setJsonPayload(httpEvent, key, data, encoding) {
```

Comment:
shouldn't you call this method only, if the response content type is actually `application/json` ?

Reply:
In general, this is a true approach, but not all APIs are always sending the right header. For example, the GitHub API returns `'content-type': 'text/plain; charset=utf-8'` with a gzip that insides there's a JSON.

Reply:
but I guess only if the `accept` request header was `application/json` ?